### ISSUE TITLE
Webhook - unlockMethod value

### ISSUE THREAD

**Issue by ThatProgrammerr** [OpenWonderLabs/SwitchBotAPI#338]
### Current Situation

I unlock my front door every time either using the Keypad Touch on my front door, to trigger the Lock Pro, or via the app. I usually use the button on the lock to exit too. Although, when looking at the webhook message for these unlocking and locking, I notice that there is no way to know if the lock was opened by any specific method. For example, if I unlock with a key, it sends a standard unlock webhook and it's as if I unlocked via a proper method.

### Proposed Change

I propose that the webhook sent should include an additional variable named 'unlockMethod', which should state the method, for example: app, keypad, manual/key. I don't mind if this is added as part of a manual call for example 'lastUnlockMethod' as I can detect the unlocking via the webhook, then ping the API myself, but that seems a bit out of the way.

### Additional Context

I utilize the webhook feature of the Switchbot API to update my internal program to keep my Home Assistant up to date on the temps from all my switchbot meters, without constantly pinging the API for updates, but I notice that this webhook lacks a feature I need.

**Comment by ThatProgrammerr** [comment]
I've gone through and found all issues relevant to this, as to see that there is a demand for this feature. I really hope the switchbot team will consider implementing something close, if not exactly what has been requested over the last few years.

**Comment by vishwajit76** [comment]
SwitchBot's product is ready good for security and authentication and not doubt on product quality.  But lack of documents , standard APIs and Webhooks it is not  useful for me. Please update the webhook events for "Keypad Touch" device. I already wait for your update.

Thanks.

**Comment by TTRSQ** [comment]
🙏

**Comment by ThatProgrammerr** [comment]
It's a real shame that feature requests like this seem to be ignored.

**Comment by koshisan** [comment]
Yeah, unfortunately Switchbot support has been utterly useless with that. We now have the third iteration of the device, this request has been there from the beginning, seems like an absolute no-brainer for any decent smart home integration (of course we want to know _who_ unlocked the door) - and yet: absolutely nothing...
However, it seems that the app at least has some way to access the protocol of the keypad. It looks like this isn't documented anywhere, but if we could find it, we could build a workaround since we do get the lock/unlock state from the lock. We could then just try to poll the device each time for the log and compare the timestamps...

**Comment by Bu66as** [comment]
Strongly supporting this request. A webhook carrying the **unlock method / which passcode or fingerprint** was used is essential for any real home-automation use case.

Concrete example: I'd like Home Assistant to send a notification like "front door unlocked by <user>" and trigger different scenes depending on who entered — impossible today because the API exposes no unlock event at all, only createKey/deleteKey.

Right now this gap forces users to add a second, separate keypad just to get per-user entry events, which defeats the purpose of the SwitchBot keypad. Please consider prioritizing this. Thanks!
